src/compiler/rust.rs

```diff
@@ -1453,8 +1454,28 @@ where
                 val.hash(&mut HashToDigest { digest: &mut m });
             }
         }
+        let mut arguments = arguments;
         // 8. The cwd of the compile. This will wind up in the rlib.
-        cwd.hash(&mut HashToDigest { digest: &mut m });
+        // This will end up in the rlib only in the case no basedir path
+        // rewriting is done and the build is operating with the absolute paths.
+        if let Ok(_idx) =
+            env_vars.binary_search_by(|(ref k, _)| k.cmp(&OsString::from("CACHEPOT_REMAP_RELATIVE")))
```

Comment:
I would assume we want this to be on by default?

Reply:
Maybe roll this out without being default for everyone and verify it on some lower-risk scenarios such as maybe CI targets that do not go into releases and such?